docs/workload-identity.md

```diff
@@ -1,14 +1,25 @@
-# Setup for Workload Identity clusters 
```

Comment:
As mentioned before, I don't think we should worry about cloud vendor specific stuff within this project.
My suggestion is to totally remove this file from this repo.